tests/test_footer.py

```diff
@@ -52,17 +57,64 @@ class Table(tables.Table):
 
 
 def test_footer_column_method():
+    class SummingColumn(tables.Column):
+        def render_footer(self, bound_column, table):
+            return sum(
+                bound_column.accessor.resolve(row) for row in table.data)
+
+    class TestTable(tables.Table):
+        name = tables.Column()
+        country = tables.Column(footer='Total:')
+        population = SummingColumn()
 
+    table = TestTable(MEMORY_DATA)
+    html = table.as_html(build_request('/'))
+
+    soup = BeautifulSoup(html, "lxml")
+    row = soup.find("tfoot").tr
+    columns = row.find_all("td")
+
+    assert columns[1].text == "Total:"
+    assert columns[2].text == "18833000"
+
+
+def test_footer_has_class():
     class SummingColumn(tables.Column):
         def render_footer(self, bound_column, table):
-            return sum(bound_column.accessor.resolve(row) for row in table.data)
+            return sum(
+                bound_column.accessor.resolve(row) for row in table.data)
 
-    class Table(tables.Table):
+    class TestTable(tables.Table):
         name = tables.Column()
         country = tables.Column(footer='Total:')
         population = SummingColumn()
 
-    table = Table(MEMORY_DATA)
+    table = TestTable(MEMORY_DATA)
     html = table.as_html(build_request('/'))
-    assert '<td>Total:</td>' in html
-    assert '<td>18833000</td>' in html
+
+    soup = BeautifulSoup(html, "lxml")
+    row = soup.find("tfoot").tr
+    columns = row.find_all("td")
+
+    assert "class" in columns[1].attrs
+
+
+def test_footer_custom_attriubtes():
+    class SummingColumn(tables.Column):
+        def render_footer(self, bound_column, table):
+            return sum(
+                bound_column.accessor.resolve(row) for row in table.data)
+
+    class TestTable(tables.Table):
+        name = tables.Column()
+        country = tables.Column(footer='Total:', attrs={'tf': {'align': 'right'}})
+        population = SummingColumn()
+
+    table = TestTable(MEMORY_DATA)
+    table.columns['country'].attrs['tf'] = {'align': 'right'}
+    html = table.as_html(build_request('/'))
+
+    soup = BeautifulSoup(html, "lxml")
+    row = soup.find("tfoot").tr
+    columns = row.find_all("td")
+    assert "align" in columns[1].attrs
```

Comment:
Having BeautifulSoup in the tests is nice, but does it allow a little more concise syntax? It would be nice to have some convenient helper in `test/utils.py`. It would also be nice to ditch the bare lxml-related stuff in the tests.

Reply:
Not being a native English speaker, I am not sure if I understood you well enough. What do you mean by "concise"? Less lines? I have compared the other option which was something along ``assert "align" in soup.find("tfoot").tr.find_all("td")[1].attrs`` and it was kind of unreadable to me. I also went with the more sparse syntax so you can actually easily plug in new people into the project and understand, what is actually happening. From my experience, you can be really concise with CSS selectors but the thing quickly becomes unreadable. So I have chosen the best thing I could come up with. If you have any other suggestion how should I re-write this code, please let me know, as I currently don't have other ideas. Maybe that's just a matter of taste.

Reply:
I'm no native either, so double distorted communication... please feel free to ask for clarification... And indeed, my comment isn't totally clear.

I meant to address that 'we' already have some mechanism in place to do what you're doing with BeautifulSoup with lxml, for example: https://github.com/jieter/django-tables2/blob/26f2fc2839314a1b111150acdd054b6ff0747e4f/tests/columns/test_general.py#L273
I'm not against adopting a new way to do it, but I prefer using one way across the tests.

Reply:
I get it now thanks for explanation. I’ll try rewriting that with lxml if it is a dependency already.